clients/src/main/java/org/apache/kafka/common/utils/KafkaLoggerFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.utils;
+
+public class KafkaLoggerFactory {
```

Comment:
Hmm.. I had considered that we might take this class in that direction over time. Perhaps we could include the log prefix for now and consider the internal map for future work? 

I think I do prefer the `LogContext` name over another smurf factory. Maybe if we have that and we also let `KafkaLogger` implement the `Logger` interface, then we have a simple interface like this:

```java
class LogContext {
  Logger logger(Class<?> clazz);
}
```

That seems nicer than the current interface.